### ISSUE TITLE
[Bug]: TypeScript plugin drops class export

### ISSUE THREAD

**Issue by marvinhagemeister** [babel/babel#14945]
### 💻

- [ ] Would you like to work on a fix?

### How are you using Babel?

Programmatic API (`babel.transform`, `babel.parse`)

### Input code

```ts
declare class Signal<T = any> {
	value: T
}

function Signal(this: Signal, value?: unknown) {
	this.value = value
}

export { Signal };
```

### Configuration

```json
{
  "presets": ["@babel/preset-typescript"]
}
```

### Current and expected behavior

Actual output:

```js
function Signal(value) {
  this.value = value;
}

export {};
```

Expected output, see this [TypeScript playground](https://www.typescriptlang.org/play?#code/CYUwxgNghgTiAEkoGdnwMoEsDmA7KEAPACrwC88UuAngHzwDeAUAJABuBAriAFzzFMAvkyYAzTrjAAXTAHtcGHPggAKKQAtMyPljwEANPA4RuAfj4SA1rlkB3XAEpGrDVoB0x7uSNcQQkSAAHgAOsjBSziy6ykIA3ExAA):

```js
function Signal(value) {
    this.value = value;
}
export { Signal };
```

Basically there is a compilation difference between babel's TypeScript preset and TypeScript itself.

### Environment

OS: macOS 12.5.1
Node: 18.8.0
Yarn: 1.22.17
@babel/core: 7.19.1
@babel/preset-typescript: 7.18.6
Monorepo: no

### Additional context

My assumption is that babel sees the `declare class Signal {}` line and tags it as a type-only thing or something like that.
